### ISSUE TITLE
command-line job submission?

### ISSUE THREAD

**Issue by jayoung** [fhdsl/FH_WDL102_Workflows#11]
The guide covers job submission via (a) the shiny app and (b) the `fhwdlR` package.

Question - is it possible to submit jobs directly from the gizmo/rhino command line?  If it's possible, that would likely be my preferred method (perhaps other users too).   Or maybe not - maybe once I get into the swing of things I'll see that's a big pain.

In any case, seems like a question worth addressing in the guide.

**Comment by jayoung** [comment]
to clarify, I meant the linux command-line!   

I notice the header of chapter 5 (on the fh.wdlR package) says it lets you submit jobs from the command-line.  Now that I've been through that material, I probably would use the R-command-line submission method: that looks very nice.  

I'm still curious though - is there a direct way, without spinning up R or using the web shiny app?

**Comment by vortexing** [comment]
The server has an API so the app and the R package are just wrappers for talking to the API.  if you start up a server, and then in your browser go to https://gizmok1:55555 (or whatever), you'll see the SWagger UI and you can futz around all you like with the endpoint definitions.  Then you can use any old command line tool to send workflows or do any of this stuff.

**Comment by jayoung** [comment]
ha - just starting to play with that UI, I can TOTALLY see why doing it the R way is more appealing.

let me see if I understand right: you would use that swagger UI to produce an example`curl` command (it also submits the workflow).   Then maybe on another occasion, you could copy and modify that `curl` command, maybe playing with additional options listed in the swaggerUI, and then just copy-paste the curl command onto the rhino/gizmo command line to submit an additional job?

**Comment by jayoung** [comment]
I had to use `http` rather than `https` in the URL to make that work (in Safari).

I'm not sure, but I suspect it might be worth a couple of sentences in the guide to say something like this:  
##
There is another way you can interact with your cromwell server, via something called a swagger interface to the Cromwell API. That lets you submit workflows directly, and/or canhelp you figure out the syntax of curl commands you could use to submit jobs via the linux command line. We won't describe it in detail here, but if you want to explore it, use your web browser to point to http://gizmok1:55555/ (or whatever node:port combination your server is running on), and play with the resulting webpage.
##
To me it's a very natural question to ask, at least for people who've been working with sbatch for a while.  Just a quick play around there helps me see how it MIGHT be done, and helps me think about whether I want to go down the command-line road, or stick with the R package and/or the shiny app.

**Comment by vortexing** [comment]
yup, these are all valid points and the main reason why I did not address the Swagger UI is that I actually don't think that We Collectively want to futz with supporting yet another interface in this instance.  I believe that documentation is in the cromwell docs and can always be shared during effective computing drop in hours, etc.  BUT I think in the beginner guide, likely the folks who'd ask about this might not be the intended audience I think.  

I am thinking maybe WDL 102 could have a reference section for advanced users and configuration guidance that would link out to existing docs for people to explore on their own.  I'm going to transfer this issue to that repo so I can address it there!!!   Regardless, it's nice to see someone else understand how useful this tool can be for a variety of types of people and preferred ways of working!  There are so many ways to use it it's easy to try to boil the ocean when documenting it in a guide. ;)
